modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java

```diff
@@ -1577,6 +1577,19 @@ protected void addImport(CodegenModel m, String type) {
         }
     }
 
+    protected void addImport(Set<String> importsToBeAddedTo, String type) {
+        if (type == null) {
+            return;
+        }
+
+        String[] parts = type.split("( [|&] )|[<>]");
```

Comment:
can you add some comments what this intends to solve?

Reply:
when this method  got composed type (`SingleFieldDependency | ConditionalSingleFieldDependency`)
I split composed type and then i added to import SingleFieldDependency and ConditionalSingleFieldDependency

Reply:
Thanks for the explanation, can you add a comment to the code?

Reply:
Could you look at updated code ?
I changed  `String[] parts = type.split("( [|&] )|[<>]");`
to `String[] parts = splitComposedType(type);`
Maybe it would be better than comment?

Reply:
looks good now